avaframe/in3Utils/generateTopo.py

```diff
@@ -0,0 +1,561 @@
+    ax.set_xlabel('along valley distance [m]')
+    ax.set_ylabel('across valley distance [m]')
+    ax.set_zlabel('surface elevation [m]')
+
+    plt.show()
+
+
+def writeDEM(z, name_ext, ncols, nrows, dx, cD):
+    """ Write topography information to file """
+
+    # Read lower left corner coordinates, cellsize and noDATA value
+    xllcorner = float(cD['xl'])
+    yllcorner = float(cD['yl'])
+    cellsize = dx
+    noDATA = float(cD['nodata_value'])
+    dem_name = cD['dem_name']
+    dpath = cD['path']
+
+    # Save elevation data to .asc file and add header lines
+    z_mat = np.matrix(z)
+    with open(os.path.join(dpath, '%s_%s_Topo.asc' % (dem_name, name_ext)), 'w') as f:
+        f.write('ncols  %d\n' % (ncols))
+        f.write('nrows  %d\n' % (nrows))
+        f.write('xllcorner  %.02f\n' % (xllcorner))
+        f.write('yllcorner %.02f\n' % (yllcorner))
+        f.write('cellsize  %d\n' % (cellsize))
+        f.write('nodata_value %.02f\n' % (noDATA))
+        for line in z_mat:
+            np.savetxt(f, line, fmt='%f')
+
+    # Log info here
+    logging.info('DEM written to: %s_%s_Topo.asc' % (dem_name, name_ext))
+
+
+def generateTopo():
+    """ Compute coordinates of desired topography with given inputs """
+
+    # Load all input Parameters
+    cfg = configparser.ConfigParser()
+    if os.path.isfile('avaframe/in3Utils/local_generateTopoCfg.ini'):
+        cfg.read('avaframe/in3Utils/local_generateTopoCfg.ini')
+    else:
+        cfg.read('avaframe/in3Utils/generateTopoCfg.ini')
+    cG = cfg['GENERAL']
+    cT = cfg['TOPO']
+    cF = cfg['FLAGS']
+    cC = cfg['CHANNEL']
+    cD = cfg['DEMDATA']
+
+    # Which DEM type
+    DEM_type = cT['DEM_type']
+
+    logging.info('DEM type is set to: %s' % DEM_type)
+    # determine number of rows and columns to define domain
+    dx = float(cG['dx'])
+    x_end = float(cG['x_end']) + dx
+    y_end = float(cG['y_end']) + dx
+    nrows = int(y_end / dx)                    # number of rows
+    ncols = int(x_end / dx)                    # number of columns
+
+    # Get parabola Parameters
+    [A, B, f_len] = getParams(cT)
+
+    # Call topography type
+    if DEM_type == 'FP':
+        [x, y, z, name_ext] = flatplane(cG, ncols, nrows, float(cT['z_elev']))
+
+    elif DEM_type == 'IP':
+        [x, y, z, name_ext] = inclinedplane(cG, ncols, nrows, cT, cC, cF)
+
+    elif DEM_type == 'HS':
+        [x, y, z, name_ext] = hockey(cG, f_len, A, B, ncols, nrows, cT, cC, cF)
+
+    elif DEM_type == 'HS2':
+        [x, y, z, name_ext] = hockeysmooth(cG, ncols, nrows, cT, cC, cF)
+
+    elif DEM_type == 'BL':
+        [x, y, z, name_ext] = bowl(cG, ncols, nrows, float(cT['r_bowl']))
+
+    elif DEM_type == 'HX':
+        [x, y, z, name_ext] = helix(cG, ncols, nrows, cT, f_len, A, B, cC, cF)
+
+    # Plot new topogrpahy
+    plotDEM(x, y, z, name_ext, cD, cG)
+
+    # Write DEM to file
+    writeDEM(z, name_ext, ncols, nrows, float(cG['dx']), cD)
```

Comment:
see issue https://github.com/avaframe/AvaFrame/issues/36 , but no need to fix here